.github/workflows/file-size-check.yml

```diff
@@ -0,0 +1,133 @@
+name: File Size Check
+# Trigger the workflow on PRs to the main branch. It builds the extension in prod mode and checks the size of the
+# vsix file and webview bundle file and compares it with the main branch. It fails if the size has increased by more than 10%.
+# For vsix, we also check if the size is greater than 25MB.
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  size-check:
+    runs-on: ubuntu-latest
+
+    steps:
+      -name: Checkout main branch
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          path: './main'
+
+      -name: Checkout PR branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: './pr'
+
+      -name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '18'
+
+      -name: Install tools
+        run: |
+            echo "Installing Yarn"
+            npm install --global yarn@1.22.19
+            echo "Installing Gulp CLI"
+            npm install --global gulp-cli@2.3.0
+            echo "Installing VSCE"
+            npm install --global vsce@2.9.2
+            echo "Installing gulp"
+            npm install --global gulp@4.0.2
+
+      -name: Build main branch
+        run: |
+            cd main
+            yarn --frozen-lockfile
+            yarn build --prod
+            yarn gulp package:online
+
+      -name: Build PR branch
+        run: |
+            cd pr
+            yarn --frozen-lockfile
+            yarn build --prod
+            yarn gulp package:online
+
+      -name: Calculate vsix file sizes
+        run: |
+            main_vsix=$(find ./main -name "*.vsix")
+            pr_vsix=$(find ./main -name "*.vsix")
+            main_size=$(stat -c%s "$main_vsix")
+            pr_size=$(stat -c%s "$pr_vsix")
+            size_diff=$((pr_size - main_size))
+            percentage_change=$((100 * size_diff / main_size))
+
+            echo "Main branch VSIX size: $main_size bytes"
+            echo "PR branch VSIX size: $pr_size bytes"
+            echo "Size difference: $size_diff bytes"
+            echo "Percentage change: $percentage_change%"
+
+            echo "main_vsix_size=$main_size" >> $GITHUB_ENV
+            echo "pr_vsix_size=$pr_size" >> $GITHUB_ENV
+            echo "vsix_percentage_change=$percentage_change" >> $GITHUB_ENV
+
+      -name: Calculate webview bundle sizes
```

Comment:
We chatted about this offline, but to capture it here: let's add some documentation with the minutia of the webview bundling and how that impacts how webview module dependencies should be consumed.